plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py

```diff
@@ -1,79 +1,52 @@
-"""Defines the templaters."""
+"""Defines the dbt templater.
+
+NOTE: The dbt python package adds a significant overhead to import.
+This module is also loaded on every run of SQLFluff regardless of
+whether the dbt templater is selected in the configuration.
+
+The templater is however only _instantiated_ when selected, and as
+such, all imports of the dbt libraries are contained within the
+DbtTemplater class and so are only imported when necessary.
+"""
 
 from collections import deque
 from contextlib import contextmanager
 import os
 import os.path
 import logging
-from typing import List, Optional, Iterator, Tuple, Any, Dict, Deque, Union
+from typing import (
+    List,
+    Optional,
+    Iterator,
+    Tuple,
+    Any,
+    Dict,
+    Deque,
+    Union,
+    TYPE_CHECKING,
+)
 
 from dataclasses import dataclass
 
-from dbt.version import get_installed_version
-from dbt.config import read_user_config
-from dbt.config.runtime import RuntimeConfig as DbtRuntimeConfig
-from dbt.adapters.factory import register_adapter, get_adapter
-from dbt.compilation import Compiler as DbtCompiler
-
-# From dbt 1.3 onwards, the default_profiles_dir resolver is
-# available. Before that version we use the flags module
-try:
-    from dbt.cli.resolvers import default_profiles_dir
-except ImportError:
-    default_profiles_dir = None
-
-# After this PR on dbt-core, we need to inject context variables
-# directly. This change was backported and so exists in some versions
-# but not others. When not present, no additional action is needed.
-# https://github.com/dbt-labs/dbt-core/pull/7949
-# On the 1.5.x branch this was between 1.5.1 and 1.5.2
-try:
-    from dbt.task.contextvars import cv_project_root
-except ImportError:
-    cv_project_root = None
-
-try:
-    from dbt.exceptions import (
-        CompilationException as DbtCompilationException,
-        FailedToConnectException as DbtFailedToConnectException,
-        DbtProjectError,
-    )
-except ImportError:
-    from dbt.exceptions import (
-        CompilationError as DbtCompilationException,
-        FailedToConnectError as DbtFailedToConnectException,
-        DbtProjectError,
-    )
-
-from dbt import flags
 from jinja2 import Environment
 from jinja2_simple_tags import StandaloneTag
 
-from sqlfluff.cli.formatters import OutputStreamFormatter
-from sqlfluff.core import FluffConfig
 from sqlfluff.core.cached_property import cached_property
 from sqlfluff.core.errors import SQLTemplaterError, SQLFluffSkipFile, SQLFluffUserError
 
 from sqlfluff.core.templaters.base import TemplatedFile, large_file_check
 
 from sqlfluff.core.templaters.jinja import JinjaTemplater
 
+if TYPE_CHECKING:  # pragma: no cover
```

Comment:
remind me what this is for?

Reply:
Ah - this imports things for when `mypy` runs, but otherwise not. That's why it's not covered in tests. https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING